schema_cache.rst

```diff
@@ -160,4 +160,84 @@ To disable auto reloading, drop the trigger:
 
 .. code-block:: postgresql
 
-  DROP EVENT TRIGGER pgrst_watch
\ No newline at end of file
+  DROP EVENT TRIGGER pgrst_watch
+
+Finer-Grained Event Trigger
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can refine the previous event trigger and only react to the events relevant to the schema cache. This also prevents unnecessary
+reloading when creating temporary tables(``CREATE TEMP TABLE``) inside functions.
+
+.. code-block:: postgresql
+
+  -- watch create and alter
+  CREATE OR REPLACE FUNCTION pgrst_ddl_watch() RETURNS event_trigger AS $$
+  DECLARE
+    obj record;
+  BEGIN
+    FOR obj IN SELECT * FROM pg_event_trigger_ddl_commands()
+    LOOP
+      IF obj.command_tag IN (
+        -- schemas
+        'CREATE SCHEMA', 'ALTER SCHEMA'
+        -- tables
+      , 'CREATE TABLE', 'CREATE TABLE AS', 'SELECT INTO', 'ALTER TABLE'
+        -- foreign tables
+      , 'CREATE FOREIGN TABLE', 'ALTER FOREIGN TABLE'
+        -- views
+      , 'CREATE VIEW', 'ALTER VIEW'
+        -- materialized views
+      , 'CREATE MATERIALIZED VIEW', 'ALTER MATERIALIZED VIEW'
+        -- functions
+      , 'CREATE FUNCTION', 'ALTER FUNCTION'
+        -- triggers
+      , 'CREATE TRIGGER'
+        -- types
+      , 'CREATE TYPE'
+        -- rules
+      , 'CREATE RULE'
+        -- comments
+      , 'COMMENT'
+      )
+      -- don't notify in case of CREATE TEMP table or other objects created on pg_temp
+      AND obj.schema_name is distinct from 'pg_temp'
+      -- don't notify on extensions changes
+      AND obj.in_extension IS FALSE
```

Comment:
No equivalent for DROPs on this one.

Reply:
I don't understand the comment.

---

Apart from this:

Why are we not notifying on extension changes? If the extension's functions are in a public schema we are exposing them - so we should update?

What if I actually deploy my API code as an extension itself?

Reply:
Ah, ok - so the way it was before, we would have reloaded when extensions object were dropped, but not when they were created? Even better to have it consistent now.